core/src/com/unciv/logic/map/tile/TileStatFunctions.kt

```diff
@@ -27,16 +27,28 @@ class TileStatFunctions(val tile: Tile) {
         localUniqueCache: LocalUniqueCache = LocalUniqueCache(false)
     ): Stats = getTileStats(tile.getCity(), observingCiv, localUniqueCache)
 
-    fun getTileStats(city: City?, observingCiv: Civilization?,
-                             localUniqueCache: LocalUniqueCache = LocalUniqueCache(false)
+    fun getTileStats(
+        city: City?, observingCiv: Civilization?,
+        localUniqueCache: LocalUniqueCache = LocalUniqueCache(false)
     ): Stats {
-        val stats = getTileStatsBreakdown(city, observingCiv, localUniqueCache).toStats()
+        val statsBreakdown = getTileStatsBreakdown(city, observingCiv, localUniqueCache)
 
-        for ((stat, value) in getTilePercentageStats(observingCiv, city, localUniqueCache)) {
-            stats[stat] *= value.toPercent()
-        }
+        val improvement = tile.getUnpillagedTileImprovement()
 
-        return stats
+        val road = if (!tile.roadIsPillaged && tile.roadStatus != RoadStatus.None)
+            tile.roadStatus.improvement(tile.ruleset)
+        else null
+
+        val percentageStats = getTilePercentageStats(observingCiv, city, localUniqueCache)
+        for (stats in statsBreakdown.filter { it.first != improvement?.name && it.first != road?.name })
+            for ((stat, value) in percentageStats["Terrain"]!!)
+                stats.second[stat] *= value.toPercent()
+        for ((stat, value) in percentageStats["Improvement"]!!)
+            statsBreakdown.firstOrNull { it.first == improvement?.name }?.second?.let { it[stat] *= value.toPercent() }
+        for ((stat, value) in percentageStats["Road"]!!)
+            statsBreakdown.firstOrNull { it.first == road?.name }?.second?.let { it[stat] *= value.toPercent() }
```

Comment:
Sounds to me like

for stats in statsBreakdown
     val percentageStats = if (stats.first == improvement?.name) percentageStats["Improvement"}
        else if (stats.first == road?.name) percentageStats["Road"] else percentageStats["Terrain"}
   for (stat, value) in percentageStats: stats.second[stat] *= value.toPercent()